### ISSUE TITLE
toolchain scripts ignore --with-mkl=no if MKLROOT variable is set for some packages

### ISSUE THREAD

**Issue by Pibemanden** [cp2k/cp2k#2175]
As the title indicates I was trying to install the current git version of cp2k on some of our local clusters that doesn't have the `c++` bindings for mkl installed. I.e. I can only compile fortran applications with MKL and therefore I have to run `./install_cp2k_toolchain.sh --install-all --with-mkl=no` to get a clean install of the toolchain libraries. That doesn't work for some libraries however, `COSMO` being the first that fails. Since I guess that cmake automatically detects my local mkl install if I have set the environment variable `MKLROOT` to point to the MKL directory.

I was eventually able to build the toolchain and cp2k successfully after unsetting this variable. So I was wondering if this is the intended behaviour, or if `--with-mkl=no` should somehow make sure that a local version of mkl is not used by accident.

**Comment by he-zilong** [comment]
> ... Since I guess that cmake automatically detects my local mkl install if I have set the environment variable `MKLROOT` to point to the MKL directory.

This behavior seems to be intentional in the `FindMKL.cmake` of [COSMA](https://github.com/eth-cscs/COSMA/blob/45507f616fd9e80e63f2975250569465a7bf6a9d/cmake/FindMKL.cmake#L145-L151) and [COSTA](https://github.com/eth-cscs/COSTA/blob/master/cmake/FindMKL.cmake#L115-L119) for instance; maybe it is better to create issues in their corresponding repositories and see if a CMake option can be added to toggle the logic (so as to be potentially included in the CP2K toolchain script for building procedures of individual libraries). Meanwhile, telling toolchain to unset the `$MKLROOT` environment variable automatically upon `--with-mkl=no` is not really an appropriate move... Until the issue is determined to be actually at CP2K's side and not the dependencies' sides, I would prefer having this ticket closed.

**Comment by Growl1234** [comment]
I agree to add `unset MKLROOT` to the regarding part of the toolchain scripts, however. For now, installing CP2K with Spack (`make_cp2k.sh`) can ensure this problem to not happen.
